docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+  That mismatch could manifest as incorrect time-slot translations eg (within ledger rules execution, but not within Node-To-Client query reponses).
+
+- (3) ZBE is allowed when ticking a ledger state; the HFC will invoke multiple translation functions in sequence.
+  But cross-era forecasting cannot skip eras, so no headers after an epoch with zero blocks could be validated, and so blocks could not progatate in the net.
+  Moreover, today's node would not even mint blocks, since the leadership check (at least currently) relies on forecasting.
+
+## Table for an Particular Hypothetical Cardano HFC
+
+The following table and associated notes classifies an hypothetical HFC implementation in terms of these dimensions.
+Nicolas Frisby anticipates this implementation would be easier to understand, less likely to surprise users, more general (ie place fewer requirements on the block types), and not require excessive code changes.
+
+| DST | CCT | PIL | DSR | ZSE | ZBE | QSR |
+| --- | --- | --- | --- | --- | --- | --- |
+| S   | D   | I   | S 2 | D   | D 1 | I 3 |
+
+- (1) ZBE would be achievable for a prefix of eras when constructing the initial HFC ledger state (in the `ProtocolInfo` passed to the Consensus layer), but otherwise the HFC as a block type does not need to support ZBE at all.
+
+- (2) Esgen points out that perhaps DSR could/should still be Insensitive, and that wouldn't obviously lose too much simplicity.
+
+- (3) Having QSR=Insensitive despite DSR=Sensitive requires an additional mechanism.
+  In this design point, that is anticipated to be constraining these queries to only be answered via an immutable ledger state.
+
+## Disclaimer
+
+There may be other points in the design space worth considering.
+(Or even other dimensions, of course.)
+In particular, DST=Insensitive and PIL=Impossible could be achieved by changing the Cardano ledger's governance to match the existing Cardano HFC logic.
+Specifically, if the ledger did not enact governance actions unless there have actually been k+1 blocks since they were ratified even despite possible Chain Growth violations, then the Cardano HFC would never override the Cardano ledger.
+
+However, Frisby thinks such block counting (and the related "double-stability" in the existing ledger rules) is strictly undue complexity that should be removed instead of propagated even deeper.
+
+---
+
+# DRAFT NUMBER 1
+
+## Introduction
+
+This document motivates a simplification of the HFC, that would also alleviate the "double stability" constraint on the ledger's governance rules.
+
+Specifically, it suggests that the HFC should not be vetoing the ledger's decision to transition to the next era while that decision is not yet immutable.
+This veto can only happen in the presence of a Chain Growth violation, but it's confusing none-the-less and requires extra implementation complexity in the HFC.
```

Comment:
Could we add an example of how the CG violation would lead to the HFC vetoing the ledger's decision? I thought that the requirement of 2-stability-windows requirement was due to rollbacks.

Reply:
2-stability-windows only suffices to prevent the veto if the chain satisfies Chain Growth.

Without the Chain Growth property, nothing prevents a rollback from rolling back (up to `k`) blocks that collectively span more than one stability window of slots. Or two stability windows. Or any number of them for that matter.